lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1980,16 +2064,206 @@ def add_existing_vmdk(self, vm_obj, expected_disk_spec, diskspec, scsi_ctl):
+            disk_ctl, disk_list = self.device_helper.get_controller_disks(vm_obj, ctl['type'], ctl['num'])
+            if disk_ctl is None:
+                # check if scsi controller key already used
+                if ctl['type'] in self.device_helper.scsi_device_type.keys():
+                    scsi_ctls = self.get_vm_scsi_controllers(vm_obj)
+                    if scsi_ctls:
+                        for scsi_ctl in scsi_ctls:
+                            if scsi_ctl.device.busNumber == ctl['num']:
+                                self.module.fail_json(msg="Specified SCSI controller number '%s' is already used by: %s"
+                                                      % (ctl['num'], scsi_ctl))
+                # create new disk controller if not exist
+                disk_ctl_spec = self.device_helper.create_disk_controller(ctl['type'], ctl['num'])
+                self.change_detected = True
+                self.configspec.deviceChange.append(disk_ctl_spec)
+            else:
+                disk_ctl_spec = vim.vm.device.VirtualDeviceSpec()
+                disk_ctl_spec.device = disk_ctl
+            for j in range(0, len(ctl['disk'])):
+                hard_disk = None
+                hard_disk_exist = False
+                disk_unit_number = ctl['disk'][j]['unit_number']
+                # from attached disk list find the specified one
+                if len(disk_list) != 0:
+                    for disk in disk_list:
+                        if disk.unitNumber == disk_unit_number:
+                            hard_disk = disk
+                            hard_disk_exist = True
+                            break
+                # if find the disk do reconfigure
+                if hard_disk_exist:
+                    hard_disk_spec = vim.vm.device.VirtualDeviceSpec()
+                    hard_disk_spec.device = hard_disk
+                    self.set_disk_parameters(hard_disk_spec, ctl['disk'][j], reconfigure=True)
+                    hard_disk_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.edit
+                    self.configspec.deviceChange.append(hard_disk_spec)
+                # if no disk or the specified one not exist do create new disk
+                if len(disk_list) == 0 or not hard_disk_exist:
+                    hard_disk = self.device_helper.create_hard_disk(disk_ctl_spec, disk_unit_number)
+                    self.set_disk_parameters(hard_disk, ctl['disk'][j])
+                    hard_disk.fileOperation = vim.vm.device.VirtualDeviceSpec.FileOperation.create
+                    self.configspec.deviceChange.append(hard_disk)
+
     def configure_disks(self, vm_obj):
         # Ignore empty disk list, this permits to keep disks when deploying a template/cloning a VM
         if len(self.params['disk']) == 0:
             return
 
-        scsi_ctl = self.get_vm_scsi_controller(vm_obj)
+        # check if do multiple types of controllers and disks configuration
```

Comment:
I tried to run the test-suite and it fails: `self.get_vm_scsi_controllers()[0]` returns `None` because the machine is not created yet.

I added this here to avoid the problem:

```python
        if not vm_obj:
            return
```

Reply:
Fixed in line #2281-#2289, thanks for testing and reviewing this.